src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      default:
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
```

Comment:
The logic for the 'clear' action type seems to be placed in the 'default' case. This is misleading as the 'default' case should be used for error handling. You should explicitly handle the 'clear' action type in a separate case and use the 'default' case to throw an error or handle unexpected action types.